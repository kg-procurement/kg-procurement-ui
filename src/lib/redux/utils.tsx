import { SerializedError } from '@reduxjs/toolkit'
import { FetchBaseQueryError } from '@reduxjs/toolkit/query'

export type ZodParseError = Extract<
  FetchBaseQueryError,
  {
    status: 'PARSING_ERROR'
    originalStatus: number
    data: string
    error: string
  }
>

export function isZodParseError(
  err: SerializedError | FetchBaseQueryError | undefined,
): err is ZodParseError {
  if (!err) return false
  return (
    'data' in err &&
    'error' in err &&
    'originalStatus' in err &&
    'status' in err
  )
}

export function extractZodError(err: ZodParseError): React.ReactNode {
  const errorList = err.error.split('\n').slice(0, -1)
  return (
    <div className="flex flex-col">
      <div className="font-semibold">
        Response does not match validator schema:
      </div>
      <ul className="list-inside list-disc">
        {errorList.map((e, i) => (
          <li key={i}>{e}</li>
        ))}
      </ul>
    </div>
  )
}

export function isErrorWithMessage(
  err: FetchBaseQueryError,
): err is { status: number, data: { error: string } } {
  if (!err.data) return false
  return (
    typeof err.data === 'object' &&
    'error' in err.data &&
    typeof err.data.error === 'string'
  )
}
