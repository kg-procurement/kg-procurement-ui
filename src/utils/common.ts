export const noop = () => {}

export const formatPrice = (
  amount: number,
  quantityPrice: number,
  currencyCode: string,
  uomName: string
): string => {
  const formattedPrice = new Intl.NumberFormat(
    currencyCode === 'IDR' ? 'id-ID' : 'en-US',
    {
      style: 'currency',
      currency: currencyCode,
    }
  ).format(amount);

  return quantityPrice > 1
    ? `${formattedPrice} / ${quantityPrice} ${uomName}`
    : `${formattedPrice} / ${uomName}`;
};
