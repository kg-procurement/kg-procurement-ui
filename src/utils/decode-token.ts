export interface Claims {
  sub: string
  exp: number
}

export function decodeToken(token: string): Claims | null {
  try {
    const payload = JSON.parse(atob(token.split('.')[1]))
    return {
      sub: payload.sub,
      exp: payload.exp,
    }
  }
  catch (error) {
    console.error('Failed to decode token:', error)
    throw error
  }
}
