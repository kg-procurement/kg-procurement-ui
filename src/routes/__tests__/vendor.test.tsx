import { render, screen } from '@testing-library/react'

import { withWrappers } from '@/lib/testing/utils.tsx'

import Page from '../vendor.tsx'

describe('<VendorPage />', () => {
  it('should render correctly without crashing', () => {
    render(withWrappers(<Page />))
    expect(screen.getByText('Search Vendor')).toBeInTheDocument()
  })
})
