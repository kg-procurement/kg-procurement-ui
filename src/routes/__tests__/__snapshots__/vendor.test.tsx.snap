// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<VendorPage/> > should render the vendors table content properly 1`] = `
"ZOD_PARSING_ERROR
Response does not match validator schema:
locations: Requiredundefined
Evaluation Form
SearchLog out
Search Vendor
Lorem Ipsum
Select Location...
undefined
Vendor NameLocationPerformance ScoreAdriza Nikmah, PTIndonesia0undefinedToko AmazonIndonesia0undefinedAneka WarnaIndonesia0undefinedAngkasa Buana Niaga Cipta, PTIndonesia0undefinedApartement Permata SenayanIndonesia0undefinedArta MuaraIndonesia0undefinedAsaba Computer CentreIndonesia1undefinedAstra Graphia TBK, PTIndonesia0undefinedAsus Service CenterIndonesia0undefinedAtrindo Asia Global, PTIndonesia0undefined
undefinedPreviousundefinedMore pages012undefinedMore pagesNext
undefined
0 of 99 row(s) selectedEmail Selected VendorÂ© 2024 KOMPAS
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
Contact us (Email):"
`;
